src/builtins/BuiltinTemporal.cpp

```diff
@@ -591,6 +613,100 @@ static Value builtinTemporalDurationPrototypeAbs(ExecutionState& state, Value th
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, "Invalid type");
+    }
+
+    if (!TemporalInstant::isValidEpochNanoseconds(argv[0].toBigInt(state))) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::RangeError, "Invalid epoch nanoseconds");
+    }
+
+    return TemporalInstant::createTemporalInstant(state, argv[0].toBigInt(state), newTarget);
+}
+
+static Value builtinTemporalInstantFrom(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    if (argc == 0) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, ErrorObject::Messages::GlobalObject_IllegalFirstArgument);
+    }
+
+    if (argv[0].isObject() && argv[0].asObject()->isTemporalInstantObject()) {
+        return TemporalInstant::createTemporalInstant(state, argv[0].asObject()->asTemporalInstantObject()->getNanoseconds());
+    }
+
+    return TemporalInstant::toTemporalInstant(state, argv[0]);
+}
+
+static Value builtinTemporalInstantFromEpochSeconds(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    return temporalInstantFromEpoch(state, argc, argv, new BigInt(TemporalInstant::secondToNanosecond));
+}
+
+static Value builtinTemporalInstantFromEpochMilliSeconds(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    return temporalInstantFromEpoch(state, argc, argv, new BigInt(TemporalInstant::millisecondToNanosecond));
+}
+
+static Value builtinTemporalInstantFromEpochMicroSeconds(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    return temporalInstantFromEpoch(state, argc, argv, new BigInt(TemporalInstant::microsecondToNanosecond));
+}
+
+static Value builtinTemporalInstantFromEpochNanoSeconds(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    return temporalInstantFromEpoch(state, argc, argv, new BigInt((int64_t)1));
+}
+
+static Value builtinTemporalInstantCompare(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    auto firstTemporalInstant = TemporalInstant::toTemporalInstant(state, argv[0]);
+    auto secondTemporalInstant = TemporalInstant::toTemporalInstant(state, argv[1]);
+    return Value(TemporalInstant::compareEpochNanoseconds(state, const_cast<BigInt&>(*(firstTemporalInstant.asObject()->asTemporalInstantObject()->getNanoseconds())), const_cast<BigInt&>(*(secondTemporalInstant.asObject()->asTemporalInstantObject()->getNanoseconds()))));
+}
+
+static Value builtinTemporalInstantPrototypeEpochSeconds(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    checkTemporalInstant(state, thisValue);
+
+    return Value(const_cast<BigInt&>(*(thisValue.asObject()->asTemporalInstantObject()->getNanoseconds())).division(state, new BigInt(TemporalInstant::secondToNanosecond)));
```

Comment:
Why did you cast a BigInt value here?
It would be possible to directly allocate a BigInt pointer value to `Value` here
Or just return a BigInt pointer.

Reply:
I'm casting the nanoseconds returned by the `TemporalInstant` object and the BigInt class doesn't work with const Objects i need to cast it. After that i call the `division()` method of the BigInt class and return it's result.

Reply:
I see.
I add an [pr #1129](https://github.com/Samsung/escargot/pull/1129) which changes BigInt operations to constant functions. So you can directly use constant BigInt pointer after this pr landed.

Reply:
Thank you!